### PR TITLE
Fix edges with ambient occlusion

### PR DESCRIPTION
**Pull request by MarcNeely**
This fixes edges when ambient occlusion is turned on so that they are not darkened by AO processing.

It does this by adding some processing to the 2nd Blur pass of AO to read the depth/order texture from the pick data so that the occlusion texture is set to the default of 1 where edges exist.  Since hidden edges are not included in the pick data (as they are not pickable) when hidden edges are turned on a 2nd depth/order texture is created to also include these edges.  In most cases the revealage texture is re-purposed for this so as not to allocate another render target/texture, but in the case of anti-aliasing the revealage texture is of the wrong type to use for multisampling so an extra texture is created.

Processing in the renderOpaque routine got more specialized for AO processing, so a seperate renderOpaqueAO routine was made to streamline the non-AO path better.

### CODE REVIEW

File: core/frontend/src/render/webgl/glsl/Blur.ts [hunk 1]
```diff
@@ -48,16 +48,32 @@ const computeBlur = `
   return result;
 `;
 
+// This optionally skips adding in the blurr texture result if the current pixel is a linear/edge/silhouette.
```
Comment (DStradley) [inline core/frontend/src/render/webgl/glsl/Blur.ts:51]: Typo on blur

**Comment by markschlosseratbentley** [comment]
@MarcNeely Overall looks good to me.  Running Image Tests on this branch ahead of time would probably be a good idea; make sure all the visual changes are expected.

File: core/frontend/src/render/webgl/glsl/Blur.ts (hunk 1 above)
Comment (MarcNeely) [reply core/frontend/src/render/webgl/glsl/Blur.ts]: fixed.

**Comment by MarcNeely** [comment]
> @MarcNeely Overall looks good to me. Running Image Tests on this branch ahead of time would probably be a good idea; make sure all the visual changes are expected.

Ran one last night & got only changes that were expected.  There weren't really any good test cases for edges and AO, so I've been getting some saved views for those to David today.

**Comment by markschlosseratbentley** [comment]
> > @MarcNeely Overall looks good to me. Running Image Tests on this branch ahead of time would probably be a good idea; make sure all the visual changes are expected.
> 
> Ran one last night & got only changes that were expected. There weren't really any good test cases for edges and AO, so I've been getting some saved views for those to David today.

@MarcNeely Great, sounds good.  Looks good to me, then.

**Review by markschlosseratbentley — APPROVED** [review]

**Comment by DStradley** [comment]
> Ran one last night & got only changes that were expected. There weren't really any good test cases for edges and AO, so I've been getting some saved views for those to David today.

@MarcNeely, the test views you gave me are all in place now and tested, and reference images for them, so you can run a new ImageTests now.

**Review by DStradley — APPROVED** [review]
We tested MSAA version as well.

**Comment by markschlosseratbentley** [comment]
@mergifyio update

**Pull request merged**